src/test/rpc/LedgerRPC_test.cpp

```diff
@@ -1222,21 +1222,40 @@ class LedgerRPC_test : public beast::unit_test::suite
     void
     testLedgerEntryUnknownOption()
     {
-        testcase("ledger_entry Request Unknown Option");
+        testcase("ledger_entry v1 Request Unknown Option");
         using namespace test::jtx;
         Env env{*this};
 
         std::string const ledgerHash{to_string(env.closed()->info().hash)};
 
         // "features" is not an option supported by ledger_entry.
         Json::Value jvParams;
+        jvParams[jss::api_version] = 1;
         jvParams[jss::features] = ledgerHash;
         jvParams[jss::ledger_hash] = ledgerHash;
         Json::Value const jrr =
             env.rpc("json", "ledger_entry", to_string(jvParams))[jss::result];
         checkErrorValue(jrr, "unknownOption", "");
     }
 
+    void
+    testLedgerEntryInvalidParams()
+    {
+        testcase("ledger_entry v2 Request Invalid Parameters");
+        using namespace test::jtx;
+        Env env{*this};
+
+        std::string const ledgerHash{to_string(env.closed()->info().hash)};
+
+        // "features" is not an option supported by ledger_entry.
+        Json::Value jvParams;
+        jvParams[jss::features] = ledgerHash;
+        jvParams[jss::ledger_hash] = ledgerHash;
+        Json::Value const jrr =
+            env.rpc("json", "ledger_entry", to_string(jvParams))[jss::result];
+        checkErrorValue(jrr, "invalidParams", "");
```

Comment:
Related to https://github.com/XRPLF/rippled/pull/4568#discussion_r1227054960, somehow this method tests version 2 even though (a) `RPC::apiVersionIfUnspecified` is 1 and (b) the request object is missing field `api_version`. How?

Reply:
a,b) It only happens in unit-tests (by default it choses max api version). Actual rippled uses v1 by default (if nothing is specified) as you'd expect. Still trying to figure out why this happens unit tests

Reply:
Looks like this is what's causing the problem: https://github.com/XRPLF/rippled/blob/beb0904a322bcaef777bedfe17c92c2d2f56e916/src/ripple/net/impl/RPCCall.cpp#L1474